### ISSUE TITLE
NET-1995 Update S4144: Rule should ignore value of literals

### ISSUE THREAD

**Issue by Evangelink** [SonarSource/sonar-dotnet#1672]
### Description

Currently the rule reports on methods with the exact same content. We think this would be valuable to change the implementation to ignore the value of literals.

For example:

```csharp
public string MyMethod()
{
    string s = "foo";
    var input = Console.ReadLine();
    return input + s;
}

public string MyOtherMethod()
{
    string s = "bar";
    var input = Console.ReadLine();
    return input + s;
}
```

There should be an issue raised to suggest to the developer to factorize the common behavior.

**Comment by andrei-epure-sonarsource** [comment]
One idea is to replace all string literal nodes with empty node and use the built-in comparison from roslyn

another idea is to rewrite from scratch

**Comment by jamieromanowski** [comment]
FWIW - If this gets done, it should not be enforced for test files where you might have to test a function with the same pattern but different strings.

**Comment by pavel-mikula-sonarsource** [comment]
This idea can be applied to any literal, not just strings.

**Comment by pavel-mikula-sonarsource** [comment]
Moved to [NET-1995](https://sonarsource.atlassian.net/browse/NET-1995)

[NET-1995]: https://sonarsource.atlassian.net/browse/NET-1995?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
